### PR TITLE
Cleanup code formatting, add Maven configuration, add CI/CD pipelines

### PR DESCRIPTION
**Pull request by vdye**
This pull request does a bunch of things, so the changes are easier to review commit-by-commit. 

- Commits 1 & 2 cleanup whitespace in the project; the first replaces tabs with spaces, the second removes trailing whitespace. **To avoid an over-complicated diff, make sure you [hide whitespace changes](https://github.blog/changelog/2021-10-14-hiding-whitespace-is-now-remembered-for-each-pull-request/).**
- Commits 3-5 reorganize files in the repository to adhere to [Apache's recommended directory layout](https://maven.apache.org/guides/introduction/introduction-to-the-standard-directory-layout.html).
- Commit 6 replaces the stray Eclipse configuration files for the project with a [Maven](https://maven.apache.org/index.html) `pom.xml`.
- Commit 7 creates a CI workflow that verifies the code compiles and the tests pass. See [here](https://github.com/vdye/rank-panda/actions/runs/5161535832) for an example of a successful workflow run. 
- Commit 8 creates a CD workflow that creates the Rank Panda `.jar` file and attaches it to a draft GitHub release. [Here's](https://github.com/vdye/rank-panda/actions/runs/5161243347) an example of a successful run, and [here's](https://github.com/vdye/rank-panda/releases/tag/v0.1.0) an example generated release.
- Commit 9 adds a `README.md` describing how to download & run Rank Panda, as well as how to develop for it. 

CC: @parker-rho (let me know in the comments if you have any questions about any of this!)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  <name>Rank Panda</name>
+  <description>Drill-writing software for rank drill</description>
+
+  <properties>
+    <revision>1.0.0-SNAPSHOT</revision>
+		<maven.compiler.source>1.8</maven.compiler.source>
+		<maven.compiler.target>1.8</maven.compiler.target>
+	</properties>
+
+	<dependencies>
```
Comment (vdye) [inline pom.xml:17]: Fix these tabs

**Comment by parker-rho** [comment]
This all looks so great thank you so much for doing this I'm going to look through it all now and I'll let you know if I encounter any issues!

**Comment by parker-rho** [comment]
How can I incorporate these changes to the code I currently have?

**Comment by vdye** [comment]
> This all looks so great thank you so much for doing this I'm going to look through it all now and I'll let you know if I encounter any issues!

Thanks! I've merged it (to help with your next question), but if you do run into problems, I'll fix them ASAP.

> How can I incorporate these changes to the code I currently have?

Now that they've been merged to the `main` branch, there are a couple ways to do this. The easiest way is, on [your fork's main page](https://github.com/parker-rho/rank-panda-2.0), click the "Sync Fork" button, then pull from your `origin/main` branch. 

Alternatively, you could add this repo as a new remote (e.g. `upstream`) in your local clone:

```ShellSession
$ git remote add upstream https://github.com/bigredbands/rank-panda.git
```

Then, in your local clone of the repository, either [`git pull upstream main`](https://git-scm.com/docs/git-fetch) _or_ [`git fetch upstream main`](https://git-scm.com/docs/git-fetch) & [`git merge upstream/main`](https://git-scm.com/docs/git-merge) into whichever branch you're working on.

**Pull request merged**